hail/python/hail/backend/backend.py

```diff
@@ -41,7 +41,38 @@ def persist_matrix_table(self, mt, storage_level):
 
     def unpersist_matrix_table(self, mt, storage_level):
         return mt
-    
+
+    @abc.abstractmethod
+    def add_reference(self, config):
+        raise NotImplementedError
```

Comment:
I think people often write either `pass` or `...` instead of `raise NotImplementedError` in these cases

Reply:
(you can't instantiate a class without overriding them anyway)

Reply:
That's not true in my experience, which is why I added the raise.

```
$ cat test.py
import abc

class A(object):
    @abc.abstractmethod
    def foo(self):
        pass

class B(A):
    def goo(self):
        return 5

b = B()
print('foo', b.foo())
print('goo', b.goo())
$ python test.py
foo None
goo 5
```

Reply:
Ah, the abstract base has to have ABCMeta as its metaclass for this to get checked.  Will fix.